kuma/static/js/utils/post-message-handler.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Beacons events of type 'Performance Events' to GA
+ * @param {Object} data - The data associated with this event
+ */
+function handlePerformanceEvents(data) {
+    // one of hitType: event
+    mdn.analytics.trackEvent({
+        category: data.category,
+        action: data.action,
+        label: new Date().getTime() + '-' + mdn.utils.randomString(5),
+        value: data.value - performance.timing.fetchStart
+    });
+
+    // one of hitType: timing
+    mdn.analytics.trackTiming({
+        category: data.category,
+        timingVar: data.action,
+        value: data.value - performance.timing.fetchStart
+    });
+}
+
+/**
+ * Processes marks and sends beacons to GA based on mark value
+ * @param {Object} perfData - Object containing performance mark/measure information
+ */
+function handlePerfMarks(perfData) {
+    // if there is no `measureName` property, just set a mark
+    if (!perfData.measureName) {
+        window.mdn.perf.setMark(perfData.markName);
+    } else {
+        window.mdn.perf.setMark(perfData.markName);
+        window.mdn.perf.setMeasure({
+            measureName: perfData.measureName,
+            startMark: perfData.startMark,
+            endMark: perfData.endMark
+        });
+
+        mdn.analytics.trackTiming({
+            category: 'RUM - Interactive Examples',
+            timingVar: perfData.measureName,
+            value: window.mdn.perf.getDuration(perfData.measureName)
```

Comment:
The [type of the timing event is an integer](https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#timingValue), but this is passing a float. We should truncate it on our side, like ``Math.round(window.mdn.perf.getDuration(perfData.measureName))``, or adjust ``get_duration`` to return an integer.

Reply:
Aha, good catch. I did not know it had this restriction/requirement. We can simply ensure that we never return a floating point in a central location.